### ISSUE TITLE
Evaluate Scratchblocks

### ISSUE THREAD

**Issue by DogeisCut** [PenguinMod/PenguinMod-Vm#24]
Another possibly dumb idea, [scratchblocks](https://scratchblocks.github.io/) eval blocks.

![msedge_xj85EsLdnp](https://github.com/PenguinMod/PenguinMod-Vm/assets/23178844/185fd758-8112-4823-ab9e-8e4fac242b43)
![msedge_SYCxaBsM8v](https://github.com/PenguinMod/PenguinMod-Vm/assets/23178844/56585a56-b7a4-4be6-9057-1309cf8d92c8)
![msedge_lt2KWgdNJo](https://github.com/PenguinMod/PenguinMod-Vm/assets/23178844/323ed124-a9db-4f87-b7a2-918c83e224a4)
![msedge_A50zTRLK1w](https://github.com/PenguinMod/PenguinMod-Vm/assets/23178844/6c60e5b4-06d4-4da5-8664-d2abe0abe8d7)

Basically it'd run the blocks you type into the input. With the reporter well, reporting, and the command, running the command blocks. It would also have access to variables and such.

Could be renamed to `evaluate stack` and `evaluate reporter` leaving out the boolean too

**Comment by lego7set** [comment]
you might wanna use the regular scratch categories from `Scratch.vm.runtime.ext_scratch3_?` and the question mark can be replaced with control, data, event, looks, motion, operators, procedures (custom blocks idk why you need this), sensing, and sound.  do your own parsing with your own custom extensions. maybe the devs could add a built in extension for this? idk.

**Comment by RedMan13** [comment]
> you might wanna use the regular scratch categories from `Scratch.vm.runtime.ext_scratch3_?` and the question mark can be replaced with control, data, event, looks, motion, operators, procedures (custom blocks idk why you need this), sensing, and sound. do your own parsing with your own custom extensions. maybe the devs could add a built in extension for this? idk.

it would be significantly easier and more compatible to just make it parse the scratchblocks code into the compiler sudo code and compile that
